### ISSUE TITLE
I have a little question about code

### ISSUE THREAD

**Issue by darlingfirebox** [muhanzhang/pytorch_DGCNN#5]
`def loop_dataset(g_list, classifier, sample_idxes, optimizer=None, bsize=cmd_args.batch_size):
    total_loss = []
    total_iters = (len(sample_idxes) + (bsize - 1) * (optimizer is None)) // bsize
    pbar = tqdm(range(total_iters), unit='batch')

    n_samples = 0
    for pos in pbar:
        selected_idx = sample_idxes[pos * bsize : (pos + 1) * bsize]

        batch_graph = [g_list[idx] for idx in selected_idx]
        _, loss, acc = classifier(batch_graph)

        if optimizer is not None:
            optimizer.zero_grad()
            loss.backward()
            optimizer.step()

        loss = loss.data.cpu().numpy()
        pbar.set_description('loss: %0.5f acc: %0.5f' % (loss, acc) )

        total_loss.append( np.array([loss, acc]) * len(selected_idx))

        n_samples += len(selected_idx)
    if optimizer is None:
        assert n_samples == len(sample_idxes)
    total_loss = np.array(total_loss)
    avg_loss = np.sum(total_loss, 0) / n_samples
    return avg_loss, acc`
This function return a acc for test_acc, but this acc was just the last batch's acc, is this a mistake? This function was in main.py

**Comment by darlingfirebox** [comment]
And i'm sorry about the codes'format, i have upload it serveral times but can't fix it

**Comment by muhanzhang** [comment]
Hi, please update to the latest version where this bug is already fixed. Thanks!

**Comment by darlingfirebox** [comment]
ok，thanks!
